module/VuFind/src/VuFind/Controller/HoldsTrait.php

```diff
@@ -92,18 +92,22 @@ public function holdAction()
             return $this->redirectToRecord('#top');
         }
 
-        // Send various values to the view so we can build the form:
-        $requestGroups = $catalog->checkCapability(
-            'getRequestGroups',
-            [$driver->getUniqueID(), $patron, $gatheredDetails]
-        ) ? $catalog->getRequestGroups(
-            $driver->getUniqueID(),
-            $patron,
-            $gatheredDetails
-        ) : [];
         $extraHoldFields = isset($checkHolds['extraHoldFields'])
             ? explode(':', $checkHolds['extraHoldFields']) : [];
 
+        // Send various values to the view so we can build the form:
+        $requestGroups = [];
+        if (in_array('requestGroup', $extraHoldFields)) {
+            $requestGroups = $catalog->checkCapability(
+                'getRequestGroups',
+                [$driver->getUniqueID(), $patron, $gatheredDetails]
+            ) ? $catalog->getRequestGroups(
+                $driver->getUniqueID(),
+                $patron,
+                $gatheredDetails
+            ) : [];
+        }
```

Comment:
You could simplify this further to eliminate the ternary operator and generally reduce number of lines:

```suggestion
        $requestGroups = [];
        $requestGroupsArgs = [$driver->getUniqueID(), $patron, $gatheredDetails];
        if (in_array('requestGroup', $extraHoldFields)
            && $catalog->checkCapability('getRequestGroups', $requestGroupsArgs)
        ) {
            $requestGroups = $catalog->getRequestGroups(...$requestGroupsArgs);
        }
```

Reply:
Good idea. I only modified it slightly from your suggestion to make the scan checks happy (it was upset about `The first expression of a multi-line control structure must be on the line after the opening parenthesis`)

Reply:
Thanks, @meganschanz! That rule changed a year or two ago and I haven't gotten used to it yet. :-)